docs/development/setup-recommended.md

```diff
@@ -283,6 +275,40 @@ for better system performance while developing, but you may need to revert to `o
 whenever you need to re-provision.
 :::
 
+::::{tab-set}
+
+:::{tab-item} Windows (WSL)
+:sync: os-windows
+
+Run the following to install the Zulip development environment and
+start it. (If Windows Firewall creates popups to block services,
+simply click **Allow access**.)
+
+```console
+$ # Install/update the Zulip development environment
+$ ./tools/provision
+$ # Enter the Zulip Python environment
+$ source /srv/zulip-py3-venv/bin/activate
+$ # Start the development server
+$ ./tools/run-dev
+```
```

Comment:
It looks like you forgot to delete the original copy with per-platform comments that was just above this.

Reply:
Looks like a misresolved merge conflict. I also moved the macOS specific troubleshooting note to the right section:

``` diff
diff --git a/docs/development/setup-recommended.md b/docs/development/setup-recommended.md
index 57a544c458..5c2f1e76b9 100644
--- a/docs/development/setup-recommended.md
+++ b/docs/development/setup-recommended.md
@@ -251,30 +251,6 @@ Checking out files: 100% (1912/1912), done.
 
 ### Step 3: Start the development environment
 
-Change into the zulip directory and tell Vagrant to start the Zulip
-development environment with `vagrant up`:
-
-```console
-$ # On Windows:
-$ cd zulip
-$ vagrant plugin install vagrant-vbguest
-$ vagrant up --provider=virtualbox
-
-$ # On macOS or Linux:
-$ cd zulip
-$ vagrant up --provider=docker
-```
-
-:::{warning}
-There is a [known upstream issue on macOS](https://chat.zulip.org/#narrow/stream/21-provision-help/topic/provision.20error.20ERR_PNPM_LINKING_FAILED/near/1649241)
-that can cause provisioning to fail with `ERR_PNPM_LINKING_FAILED` or other errors. The temporary
-fix is to open the Docker desktop app's settings panel, and choose `osxfs (legacy)` under "Choose
-file sharing implementation for your containers." Once Docker restarts, you should be able to
-successfully run `vagrant up --provider=docker`. Back in Docker, you can return to using VirtioFS
-for better system performance while developing, but you may need to revert to `osxfs (legacy)`
-whenever you need to re-provision.
-:::
-
 ::::{tab-set}
 
 :::{tab-item} Windows (WSL)
@@ -334,6 +310,16 @@ $ cd zulip
 $ vagrant up --provider=docker
 ```
 
+:::{warning}
+There is a [known upstream issue on macOS](https://chat.zulip.org/#narrow/stream/21-provision-help/topic/provision.20error.20ERR_PNPM_LINKING_FAILED/near/1649241)
+that can cause provisioning to fail with `ERR_PNPM_LINKING_FAILED` or other errors. The temporary
+fix is to open the Docker desktop app's settings panel, and choose `osxfs (legacy)` under "Choose
+file sharing implementation for your containers." Once Docker restarts, you should be able to
+successfully run `vagrant up --provider=docker`. Back in Docker, you can return to using VirtioFS
+for better system performance while developing, but you may need to revert to `osxfs (legacy)`
+whenever you need to re-provision.
+:::
+
 ```{include} setup/vagrant-up.md
 
 ```
```